salt/state.py

```diff
@@ -1023,7 +1026,10 @@ def _check_cmd(cmd):
                     log.warning(ret["comment"])
                     return ret
 
+                get_return = entry.pop("get_return", None)
                 result = self._run_check_function(entry)
+                if get_return:
+                    result = salt.utils.data.traverse_dict_and_list(result, get_return)
                 if self.state_con.get("retcode", 0):
                     _check_cmd(self.state_con["retcode"])
                 elif result:
```

Comment:
I still think it's very questionable even with that patch. 

For example, we can have a string "false" in the result and it will be evaluated as True since it's non-empty string.

I think to make it really useful we need a return check, like and eval or assert.
Also, we may need a multiple checks actually, for example:

```
    test:
      test.nop:
        - name: foo
        - onlyif:
          - fun: consul.get
            consul_url: http://127.0.0.1:8500
            key:  not-existing
          - get_return: res
            when:
              - type(res) is dict
              - res.get('Result', False) is True             
```

Reply:
This is an issue. There isn't a standard for execution module return success, so parsing possible results would really mean recreating the assert structure of [saltcheck](https://docs.saltstack.com/en/master/ref/modules/all/salt.modules.saltcheck.html) where you would need to describe text equals/doesn't equal, exists/doesn't exist, is true/is false, etc. 

While very possible, I think that might be a bit much for the definition of requisites as advanced parsing should really occur in states. If it's agreed that more advanced logic should be part of the requisite system here, I'm happy to add that to the PR as well. Heck, we could just use saltcheck for the test itself and only have unless/onlyif just have to parse saltcheck's Pass/Fail output:
```
test:
  test.nop:
    - name: foo
    - unless:
      - fun: saltcheck.run_test
        test:
          module_and_function: consul.get
          assertion: assertTrue
          assertion_section: res
```

Reply:
that is actually interesting!

Reply:
We have an initiative right now to start standardizing returns from execution modules. Though it's going to take some time to complete those efforts. We're aiming to get some validation in place during MG that can start warning of non-standard returns.